spark/src/main/java/org/apache/iceberg/actions/CreateAction.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import java.util.Map;
+
+interface CreateAction extends Action<Long> {
+
+  /**
+   * Adds additional properties to the newly created Iceberg Table. Any properties with
+   * the same key name will be overwritten.
+   * @param properties a map of properties to be included
+   * @return this for chaining
+   */
+  CreateAction withAdditionalProperties(Map<String, String> properties);
+
+  /**
+   * Adds an additional property to the newly created Iceberg Table. Any properties
+   * with the same key name will be overwritten.
+   * @param key the key of the property to add
+   * @param value the value of the property to add
+   * @return this for chaining
+   */
+  CreateAction withAdditionalProperty(String key, String value);
+
+  /**
+   * Changes the name of the new Iceberg table we are making to the name passed
+   * to this method.
+   * @param newName The new name of the table to be made
+   * @return this for chaining
+   */
+  CreateAction as(String newName);
```

Comment:
`Actions.migrate` returns a `CreateAction`, but `as` seems to change the operation from an in-place migration to something unclear. Is the original table retained if the name doesn't conflict?

Similarly, I don't think there is a need for this with `Actions.snapshot` because snapshot accepts a destination table name. When I suggested `as`, my intent was to use it as a way to pass the destination table name for snapshot. But if you can't create a `SnapshotAction` without a table name we don't need this.

Reply:
I can remove it, for snapshot it doesn't make sense but for Migrate it allowed you to use the same location for data as the original table but using a different catalog identifier, it wouldn't remove the old identifier if it didn't match. That may be of limited utility so I'll drop it.